### ISSUE TITLE
[DataGrid] The checkbox shouldn't be tabbable

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#1430]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

The select all checkbox is in the tab order

## Expected Behavior 🤔

The select all checkbox is not in the tab order

## Steps to Reproduce 🕹

1. Open https://material-ui.com/components/data-grid/#mit-version

## Context

This is a follow-up on https://github.com/mui-org/material-ui-x/pull/1289#issuecomment-818619218. We missed a `tabIndex` one and test cases.

## Your Environment 🌎

v4.0.0-alpha.25

**Comment by dtassone** [comment]
Why shouldn't it be in tabbable? It's a control with a button behind it 🤔

**Comment by DanailH** [comment]
We removed the `tabIndex` from all other checkboxes as it hijacks the keyboard control. basically using the keyboard to navigate to a cell with a checkbox and hitting `enter` or `space` should select that checkbox. At least that is how I understood it.

**Comment by oliviertassinari** [comment]
> Why shouldn't it be in tabbable?

@dtassone Because this is described in the WAI-ARIA authoring practices. The grid is a composite widget. They reiterate what this notion means:

> A grid is a composite widget so it:
> - Always contains multiple focusable elements.
> - **Only one of the focusable elements contained by the grid is included in the page tab sequence** (tabbable).
> - Requires the author to provide code that manages focus movement inside it. 

https://www.w3.org/TR/wai-aria-practices/#grid

Now, the cell > checkbox case falls under this section: 

> When the above grid navigation keys move focus, whether the focus is set on an element inside the cell or the grid cell depends on cell content. See [Whether to Focus on a Cell or an Element Inside It](https://www.w3.org/TR/wai-aria-practices/#gridNav_focus).

In theory, we should have the checkbox tabbable and focused, not the cell. This is because a checkbox doesn't have any key handler that conflicts with the grid arrow navigation. In practice, I'm not sure it's manageable.
